ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -314,9 +327,28 @@ ByteBuffer getBytes() {
   @Override
   public String toString() {
     return "Offset: " + offset + ", Size: " + getSize() + ", Deleted: " + isFlagSet(Flags.Delete_Index)
-        + ", ExpiresAtMs: " + getExpiresAtMs() + ", Original Message Offset: " + getOriginalMessageOffset() + (
-        version != PersistentIndex.VERSION_0 ? (", OperationTimeAtSecs " + getOperationTimeInMs() + ", AccountId "
-            + getAccountId() + ", ContainerId " + getContainerId()) : "");
+        + ", TTL Updated: " + isFlagSet(Flags.Ttl_Update_Index) + ", ExpiresAtMs: " + getExpiresAtMs()
+        + ", Related Message Offset: " + getRelatedMessageOffset() + (version != PersistentIndex.VERSION_0 ? (
+        ", OperationTimeAtSecs " + getOperationTimeInMs() + ", AccountId " + getAccountId() + ", ContainerId "
+            + getContainerId()) : "");
+  }
+
+  /**
+   * @return the version of this {@link IndexValue}
+   */
+  short getVersion() {
+    return version;
+  }
+
+  void setExpiresAtMs(long expiresAtMsLocal) {
+    // if expiry in secs > Integer.MAX_VALUE, treat it as permanent blob
+    if (TimeUnit.MILLISECONDS.toSeconds(expiresAtMsLocal) > Integer.MAX_VALUE) {
+      expiresAtMs = Utils.Infinite_Time;
+    } else if (expiresAtMsLocal != Utils.Infinite_Time && expiresAtMsLocal < 0) {
```

Comment:
Actually, that's exactly what I want. `Utils.getTimeInMsToTheNearestSec()` handles `Infinite_Time`. 

The first if check is ok - the output of `TimeUnit.MILLISECONDS.toSeconds(expiresAtMsLocal)` on negative values (or `Infinite_Time`) is not important for that check and it will remain correct.

Side note: I had the same trouble and confusion when writing this code about correctly handling these things (negative/infinite time). I went for the most concise (and correct) code that I could think of. The semantics around this can be improved and made less confusing (but this PR is already large so I didn't want to go down that path)